bigquery_etl/cli/backfill.py

```diff
@@ -0,0 +1,222 @@
+"""bigquery-etl CLI backfill command."""
+
+import re
+import sys
+import tempfile
+from datetime import datetime
+from pathlib import Path
+
+import click
+import yaml
+
+from ..backfill.parse import (
+    BACKFILL_FILE,
+    DEFAULT_REASON,
+    DEFAULT_WATCHER,
+    TODAY,
+    Backfill,
+    BackfillStatus,
+)
+from ..backfill.validate import (
+    validate_duplicate_entry_dates,
+    validate_file,
+    validate_overlap_dates,
+)
+from ..cli.utils import paths_matching_name_pattern, project_id_option, sql_dir_option
+
+QUALIFIED_TABLE_NAME_RE = re.compile(
+    r"(?P<project_id>[a-zA-z0-9_-]+)\.(?P<dataset_id>[a-zA-z0-9_-]+)\.(?P<table_id>[a-zA-z0-9_-]+)"
+)
+
+
+@click.group(help="Commands for managing backfills.")
+@click.pass_context
+def backfill(ctx):
+    """Create the CLI group for the backfill command."""
+    # create temporary directory generated content is written to
+    # the directory will be deleted automatically after the command exits
+    ctx.ensure_object(dict)
+    ctx.obj["TMP_DIR"] = ctx.with_resource(tempfile.TemporaryDirectory())
+
+
+@backfill.command(
+    help="""Create a new backfill entry in the backfill.yaml file.  Create
+    a backfill.yaml file if it does not already exist.
+
+    Examples:
+
+    \b
+    ./bqetl backfill create moz-fx-data-shared-prod.telemetry_derived.deviations_v1 \\
+      --start_date=2021-03-01 \\
+      --end_date=2021-03-31 \\
+      --exclude=2021-03-03 \\
+    """,
+)
+@click.argument("qualified_table_name")
+@sql_dir_option
+@click.option(
+    "--start_date",
+    "--start-date",
+    "-s",
+    help="First date to be backfilled. Date format: yyyy-mm-dd",
+    type=click.DateTime(formats=["%Y-%m-%d"]),
+    required=True,
+)
+@click.option(
+    "--end_date",
+    "--end-date",
+    "-e",
+    help="Last date to be backfilled. Date format: yyyy-mm-dd",
+    type=click.DateTime(formats=["%Y-%m-%d"]),
+    default=datetime.today(),
+)
+@click.option(
+    "--exclude",
+    "-x",
+    multiple=True,
+    help="Dates excluded from backfill. Date format: yyyy-mm-dd",
+    type=click.DateTime(formats=["%Y-%m-%d"]),
+    default=None,
+)
+@click.option(
+    "--watcher",
+    "-w",
+    help="Watcher of the backfill (email address)",
+    default=DEFAULT_WATCHER,
+)
+@click.pass_context
+def create(
+    ctx,
+    qualified_table_name,
+    sql_dir,
+    start_date,
+    end_date,
+    exclude,
+    watcher,
+):
+    """CLI command for creating a new backfill entry in backfill.yaml file.
+
+    A backfill.yaml file will be created if it does not already exist.
+    """
+    try:
+        match = QUALIFIED_TABLE_NAME_RE.match(qualified_table_name)
+        project_id = match.group("project_id")
+        dataset_id = match.group("dataset_id")
+        table_id = match.group("table_id")
```

Comment:
This would make a nice util

Reply:
I will add a comment to address this at the next iteration